apps/src/p5lab/poetry/PoemSelector.jsx

```diff
@@ -1,21 +1,98 @@
-import React from 'react';
+import React, {useState} from 'react';
 import PropTypes from 'prop-types';
 import {connect} from 'react-redux';
+import StylizedBaseDialog, {
+  FooterButton
+} from '@cdo/apps/componentLibrary/StylizedBaseDialog';
 import {setPoem} from '../redux/poetry';
 import msg from '@cdo/poetry/locale';
 import {APP_WIDTH} from '../constants';
 import {POEMS} from './constants';
 
+function CreatePoemModal(props) {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [poem, setPoem] = useState('');
+
+  const body = (
+    <div>
+      <div style={styles.modalContainer}>
+        <label style={styles.label}>{`${msg.title()}: `}</label>
+        <input
+          style={styles.input}
+          value={title}
+          onChange={event => setTitle(event.target.value)}
+        />
+      </div>
+      <div style={styles.modalContainer}>
+        <label style={styles.label}>{`${msg.author()}: `}</label>
+        <input
+          style={styles.input}
+          value={author}
+          onChange={event => setAuthor(event.target.value)}
+        />
+      </div>
+      <div style={styles.modalContainer}>
+        <label style={styles.label}>{`${msg.poem()}: `}</label>
+        <textarea
+          style={styles.input}
+          value={poem}
+          onChange={event => setPoem(event.target.value)}
+        />
+      </div>
+    </div>
+  );
+
+  const footerButton = (
+    <FooterButton
+      text={msg.save()}
+      onClick={() =>
+        props.handleClose({title, author, lines: poem.split('\n')})
+      }
+      key="confirm"
+      type="confirm"
+    />
+  );
+
+  return (
+    <StylizedBaseDialog
+      body={body}
+      title={msg.editCustomPoem()}
+      isOpen={props.isOpen}
+      handleClose={props.handleClose}
+      renderFooter={() => footerButton}
+    />
+  );
+}
+
+CreatePoemModal.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
+  handleClose: PropTypes.func.isRequired
+};
+
 function PoemSelector(props) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleClose = poem => {
+    props.onChangePoem(poem);
+    setIsOpen(false);
+  };
+
   const onChange = e => {
     const poemTitle = e.target.value;
-    const poem = Object.values(POEMS).find(poem => poem.title === poemTitle);
-    if (poem) {
-      props.onChangePoem(poem);
+
+    if (poemTitle === msg.enterMyOwn()) {
+      setIsOpen(true);
+    } else {
+      const poem = Object.values(POEMS).find(poem => poem.title === poemTitle);
+      if (poem) {
+        props.onChangePoem(poem);
+      }
     }
   };
   return (
     <div style={styles.container}>
+      <CreatePoemModal isOpen={isOpen} handleClose={handleClose} />
```

Comment:
Small naming nit: Having "CreatePoemModal" start with a verb tricks my brain in a weird way, but I'm not sure there's a better name there other than something like "Poem Creation Modal".